addon/system/store/finders.js

```diff
@@ -77,6 +82,7 @@ export function _findHasMany(adapter, store, internalModel, link, relationship)
   promise = _guard(promise, _bind(_objectIsAlive, internalModel));
 
   return promise.then(function(adapterPayload) {
+    Ember.assert("You made a `findHasMany` request for a " + internalModel.modelName + "'s `" + relationship.key + "` relationship, using link " + link + ", but the adapter's response did not have any data", payloadIsNotBlank(adapterPayload));
```

Comment:
@bmac I expected this payload would look like:

```js
{ comments: [] }
```

which would make 

```js
{}
```

invalid (and what we're asserting against.

However, [this test](https://github.com/emberjs/data/blob/1d5dfa142523a4aefdde182262923a42ccb56286/tests/integration/relationships/has-many-test.js#L347) mocks the response as `[]`.

Why isn't it `{ comments: [] }`? What should this assertion be checking for?